packages/babel-plugin-proposal-pipeline-operator/src/hackVisitor.ts

```diff
@@ -1,8 +1,27 @@
 import { types as t } from "@babel/core";
+import type { NodePath, Visitor } from "@babel/traverse";
 
-const topicReferenceReplacementVisitor = {
-  TopicReference(path) {
-    path.replaceWith(t.cloneNode(this.topicVariable));
+const topicReferenceVisitor: Visitor<{
+  topicReferences: NodePath<t.TopicReference>[];
+  sideEffectsBeforeFirstTopicReference: boolean;
+}> = {
+  exit(path, state) {
+    if (path.isTopicReference()) {
+      state.topicReferences.push(path);
+    } else {
+      if (
+        state.topicReferences.length === 0 &&
+        !state.sideEffectsBeforeFirstTopicReference &&
+        !path.isPure()
+      ) {
+        state.sideEffectsBeforeFirstTopicReference = true;
+      }
+    }
+  },
+  "ClassBody|Function"(_, state) {
```

Comment:
Class body does not encapsulate all expressions under class, for example: class decorators.

Reply:
That's exactly why I used ClassBody: class decorators evaluation is not deferred!

Reply:
The problem is that exactly determining what runs "now" is hard. The list includes (but is not limited to):
- class keys
- static fields
- iifes

To avoid making the transform too complex, I'd prefer to leave it as-is and deoptimize when they happen.